python/ray/train/backends/torch.py

```diff
@@ -144,3 +145,144 @@ def on_shutdown(self, worker_group: WorkerGroup,
 
         worker_group.execute(
             shutdown_torch, destroy_process_group=len(worker_group) > 1)
+
+
+if torch:
+    from torch.nn.parallel import DistributedDataParallel
+    from torch.utils.data import DistributedSampler, DataLoader, \
+        IterableDataset
+
+    def _get_torch_device() -> torch.device:
+        rank = train.local_rank()
+
+        if torch.cuda.is_available():
+            device = torch.device("cpu")
+        else:
+            device = torch.device(f"cuda:{rank}")
+
+        return device
+
+    class _WrappedDataLoader(DataLoader):
+        def __init__(self, base_dataloader: DataLoader, device: torch.device):
+
+            self.__dict__.update(getattr(base_dataloader, "__dict__", {}))
+            self._dataloader = base_dataloader
+            self._device = device
+
+        def __len__(self):
+            return len(self._dataloader)
+
+        @property
+        def device(self) -> Optional[torch.device]:
+            return self._device
+
+        def __iter__(self):
+            iterator = iter(self._dataloader)
+            if self._device is None:
+                yield from iterator
+
+            for item in iterator:
+                yield (i.to(self.device) for i in item)
+
+    def prepare(
```

Comment:
Name this `prepare_module`/`prepare_model`?

Reply:
I had this as an arbitrary `prepare` to support optimizer preparation, etc. in the future.

Reply:
Hmm I'm a little confused of what the end state is here. Are we still planning to do something like: `prepare (*args, TorchPrepareConfig)`? Why is `prepare_dataloader` separate?

Reply:
Agreed offline- changed this to `prepare_model`